### PR TITLE
Add Rector in archive creation

### PR DESCRIPTION
**Pull request by AlexP11223**
**Please check if the PR fulfills these requirements**
- [x] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** 

feature

Now the package creation workflow also runs [Rector](https://github.com/rectorphp/rector) if the config is present. This was needed for researching PHP 8 transpiling into PHP 7.4, but also can be useful for other tasks.

Increased the timeout to 10 minutes because sometimes it takes a bit more than 5 minutes. (Rector takes around 40 seconds in the WL project)

### CODE REVIEW

**Review by Chrico — COMMENTED** [review]
What is the use case of Rector? Isn't that something you only execute once and when the project is upgraded you can remove it? 👀 Right now, it seems that code is changed on the fly, but never tested before creating a release. So you build a release on something untested?

**Comment by AlexP11223** [comment]
There are other PRs with workflows for testing the archive. :) (+ hopefully Playwright in the future)

For transpiling the idea is that we use the latest PHP for development, and then transform into 7.4 for release.
https://inpsyde.atlassian.net/browse/PROD-171?focusedCommentId=298937

**Comment by Biont** [comment]
>So you build a release on something untested?

This here is a build/packaging workflow, not a release workflow. The produced builds are a prerequisite for further (automated&manual) testing by QA.

File: .github/workflows/build-plugin-archive.yml [hunk 1]
```diff
@@ -223,7 +234,11 @@ jobs:
           sed -i "s/'__composer_autoload_files'/\'__composer_autoload_files_${{ github.sha }}'/g" "build/vendor/composer/autoload_real.php"
       - name: Move unchanged code to build directory
         if: steps.composer-tools.outputs.php-scoper != '0'
-        run: mv * build 2>/dev/null
+        run: |
+          shopt -s extglob
+          shopt -s dotglob
```
Comment (tyrann0us) [inline .github/workflows/build-plugin-archive.yml:239]: Let's simplify this.
```suggestion
          shopt -s extglob dotglob
```

File: docs/archive-creation.md [hunk 2]
```diff
@@ -7,10 +7,11 @@ To achieve that, the reusable workflow:
...
 3. Executes `inpsyde/wp-translation-downloader` if required & configured by the package
-4. Re-installs dependencies without dev-dependencies
-5. Sets current commit hash and plugin version in the plugin's main file
-6. Runs `wp dist-archive` to create the final archive (with builtin support for a `.distignore` file)
-7. Uploads it as an artifact for download or further processing
+4. Executes `rector` if configured by the package
+5. Re-installs dependencies without dev-dependencies
+6. Sets current commit hash and plugin version in the plugin's main file
+7. Runs `wp dist-archive` to create the final archive (with builtin support for a `.distignore` file)
+8. Uploads it as an artifact for download or further processing
```
Comment (tyrann0us) [inline docs/archive-creation.md:14]: The base PR, #134, does not mention PHP-Scoper here. Let's do so in this PR according to the Boy Scout rules.
```suggestion
4. Executes PHP-Scoper if configured by the package
5. Executes Rector if configured by the package
6. Re-installs dependencies without dev-dependencies
7. Sets current commit hash and plugin version in the plugin's main file
8. Runs `wp dist-archive` to create the final archive (with builtin support for a `.distignore` file)
9. Uploads it as an artifact for download or further processing
```

**Review by tyrann0us — APPROVED** [review]
Thank you for working on this. While the changes LGTM code-wise, the addition of Rector to the reusable workflows can only be temporary until all products require modern PHP versions natively, IMO.
Left two minor comments that don't stop me from approving. Thanks!

**Review by tyrann0us — APPROVED** [review]
Thanks for incorporating the changes!

**Comment by AlexP11223** [comment]
> the addition of Rector to the reusable workflows can only be temporary until all products require modern PHP versions natively, IMO.

I think products will never use only the latest version unless something greatly changes in WP/PHP world. We will probably still need to support PHP 8 when PHP 9 will be available because many merchants are slow to upgrade.

Also Rector is quite generic tool, it just transforms the code via AST using the specified rules, so maybe we find some other use cases for it.

**Pull request merged**